midp/midp.js

```diff
@@ -541,28 +500,72 @@ Native["com/sun/midp/util/ResourceHandler.loadRomizedResource0.(Ljava/lang/Strin
     return bytes;
 };
 
-MIDP.ScreenHeight = 0;
-MIDP.ScreenWidth = 0;
+MIDP.isVKVisibleAutosize = function() {
+    var expectedHeightWithNoKeyboard = window.outerHeight - MIDP.verticalChrome;
+    if (window.innerHeight == expectedHeightWithNoKeyboard) {
+        return false;
+    } else if (window.innerHeight < expectedHeightWithNoKeyboard) {
+        return true;
+    } else {
+        console.warn("window is taller than expected in isVKVisible!");
+        return false;
+    }
+};
+
+MIDP.isVKVisibleDebug = function() {
```

Comment:
Nit: it wasn't immediately obvious to me that this is the implementation of isVKVisible when autosize is disabled, so it'd help to have a comment here explaining that.